### PR TITLE
Add reservation card 

### LINKED ISSUE #21: Add reservation card (effort 4)
Once Hotel Card is finished we want to make a resevation.
We click 'Select' and it should navigate to /hotels/{city}/{hotelId}

Then we should get all available rooms for selected hotel and display them (you can create additionl component to present room).

Room should have button 'Make Reservation'

after clicking it should navigate us to /hotels/{hotelId}/{roomId}reservation

dont make this view (just navigate)

NOTE: if user is not logged in we shouldnt enable making reservation. Instead add alert with message: "Please log in to make reservation". If user is admin or hotelOwner you should display alert("You are not allowed to make reservation")

### PR DESCRIPTION
**Pull request by ibednorz**
Screenshot 2021-03-24 at 09 45 41

### CODE REVIEW

File: src/content/LoadingIcon.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import CircularProgress from '@material-ui/core/CircularProgress'
+
+export default function LoadingIcon(props) {
+  return (
+    <div className="d-flex justify-content-center">
+      <CircularProgress />
+      <span className="sr-only">Ładowanie...</span>
```
Comment (pdybowski) [inline src/content/LoadingIcon.js]: Loading...

File: src/index.js [hunk 2]
```diff
@@ -47,7 +47,7 @@ ReactDOM.render(
 
         {/* GENERAL ROUTES */}
         {generalRoutes}
-
+        <Route path="/hotels/:city/:id" component={HotelMoreDetails} />
```
Comment (pdybowski) [inline src/index.js]: Remove it.. Don't you see the Convention? xd

File: src/components/HotelMoreDetails.jsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+export function HotelMoreDetails(props) {
+  const classes = useStyles()
+  const [hotel, setHotel] = useState(null)
+  const [loading, setLoading] = useState(true)
+
+  const hotelId = props.match.params.id
+
+  const getHotel = async () => {
+    const hotel = await fetchData(
```
Comment (pdybowski) [inline src/components/HotelMoreDetails.jsx]: Try catch
And before try setLoading(true)

File: src/components/HotelMoreDetails.jsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  },
+  header: {
+    fontSize: '22px',
+  },
+}))
+
+export function HotelMoreDetails(props) {
+  const classes = useStyles()
+  const [hotel, setHotel] = useState(null)
+  const [loading, setLoading] = useState(true)
```
Comment (pdybowski) [inline src/components/HotelMoreDetails.jsx]: Should be false

File: src/components/HotelMoreDetails.jsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    <Container maxWidth="lg">
+      <div className={classes.root}>
+        <Grid item xs={4}>
+          <p className={classes.header}>{hotel.name}</p>
+          <p>{hotel.localization.city}</p>
+        </Grid>
+        <Grid item xs={6}>
+          {hotel.rooms.map((room) => {
+            return (
+              <p>
```
Comment (pdybowski) [inline src/components/HotelMoreDetails.jsx]: Make it in separate component

**Review by pdybowski — CHANGES_REQUESTED** [review]
Paste some images in the PR description

File: src/components/HotelMoreDetails.jsx (hunk 3 above)
Comment (ibednorz) [reply src/components/HotelMoreDetails.jsx]: when I set setLoading to false and then want to change it before try I get the error

File: src/components/HotelMoreDetails.jsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+      const hotel = await fetchData(
+        global.API_BASE_URL + `api/hotels/${hotelId}`,
+        'GET'
+      )
+      setHotel(hotel)
+      setLoading(false)
+    } catch (err) {}
+  }
+
+  useEffect(() => {
```
Comment (pdybowski) [inline src/components/HotelMoreDetails.jsx]: useMemo lol
and
image

Comment (pdybowski) [reply src/components/HotelMoreDetails.jsx]: and then loadingFalse will work

File: src/components/HotelMoreDetails.jsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+  const getHotel = async () => {
+    try {
+      const hotel = await fetchData(
+        global.API_BASE_URL + `api/hotels/${hotelId}`,
+        'GET'
+      )
+      setHotel(hotel)
+      setLoading(false)
+    } catch (err) {}
```
Comment (pdybowski) [inline src/components/HotelMoreDetails.jsx]: set alert - look at Login component

File: src/components/HotelMoreDetails.jsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    fontSize: '22px',
+  },
+}))
+
+export function HotelMoreDetails(props) {
+  const classes = useStyles()
+  const [hotel, setHotel] = useState(null)
+  const [loading, setLoading] = useState(true)
+
+  const hotelId = props.match.params.id
```
Comment (pdybowski) [inline src/components/HotelMoreDetails.jsx]: lol just pass hotelId and city as props - { hotelId, city } , dont need to do that like you did

**Review by pdybowski — CHANGES_REQUESTED** [review]
image



more reqiests in comments

**Comment by pdybowski** [comment]
NEW LOOK:
image


moreover when nologgedin user clicks 'Reserve' show alert('Please log in first to make your reservation')

**Pull request merged**